opentelemetry-api/setup.cfg

```diff
@@ -54,7 +54,7 @@ where = src
 opentelemetry_context =
     contextvars_context = opentelemetry.context.contextvars_context:ContextVarsRuntimeContext
 opentelemetry_tracer_provider =
-    default_tracer_provider = opentelemetry.trace:_DefaultTracerProvider
+    default_tracer_provider = opentelemetry.trace:_NoOpTracerProvider
```

Comment:
Not completely sure about this change... Should we add a new entry point named `no_op_tracer_provider` instead?

Reply:
we could, I'm happy to go either ways

Reply:
I think it would be better to have a new entry point, which the default one is is a matter better left to configuration (environment variables and such) and not to entry points.

Reply:
Where is this (or the proposed noop) entry point used? If all this does is allow other components to discover a Noop implementation from the API package, wouldn't it be simpler for them to just encode knowledge about API's NOOP implementation directly? So anyone requiring a noop implemenation could just import `from opentelemetry.trace import NoopTracerProvider`. 

Entry point would be useful if we had multiple API packages and users had to choice to pick one. In that case entry point would allow a "pluggable" API package to provide these implementations as "plugins" but since API is not a pluggable/replaceable package, I'm not sure what purpose a default/noop entry point is serving here but may be I'm missing something really obvious.

Reply:
I searched both core and contrib repos and could not find anything referring to this entry point.

Reply:
It is used here https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/util/_providers.py#L40

Reply:
Thanks @lonewolf3739. So API is the package that consumes this entry point and we don't expect any other package to publish a default/noop implementation. Sounds like API package should just implicitly know what it wants to use as the default/noop instead of publishing an entry point itself and consuming it via packaging APIs.

Reply:
I don't have strong feelings on this one. Please open an issue if you'd like to remove the default_* entrypoints.

It's used as a fallback in the api wheren @lonewolf3739 linked, i don't know if there would be any other issues if we wanted to load the NoOpTracerProvider directly there instead. I seem to recall circular dependencies in the past, but i could be mistaken or my thinking is possibly outdated here.

Reply:
Opened #2382 to track this